src/core/document.js

```diff
@@ -1724,6 +1724,22 @@ class PDFDocument {
     if (field.has("T")) {
       const partName = stringToPDFString(await field.getAsync("T"));
       name = name === "" ? partName : `${name}.${partName}`;
+    } else {
+      let obj = field;
+      while (true) {
+        obj = await obj.getAsync("Parent");
+        if (obj instanceof Ref) {
+          obj = await xref.fetchAsync(obj);
+        }
```

Comment:
Isn't this block redundant, since the `getAsync` call above should already resolve References?